### PR TITLE
support secure connection

### PR DESCRIPTION
**Pull request by BusyJay**

### CODE REVIEW

File: grpc-sys/src/lib.rs [hunk 1]
```diff
@@ -75,6 +75,30 @@ pub enum GrpcStatusCode {
...
+    fn from(code: i32) -> GrpcStatusCode {
+        match code {
+            0 => GrpcStatusCode::Ok,
+            1 => GrpcStatusCode::Cancelled,
+            3 => GrpcStatusCode::InvalidArgument,
+            4 => GrpcStatusCode::DeadlineExceeded,
+            5 => GrpcStatusCode::NotFound,
+            6 => GrpcStatusCode::AlreadyExists,
+            7 => GrpcStatusCode::PermissionDenied,
+            16 => GrpcStatusCode::Unauthenticated,
```
Comment (siddontang) [inline grpc-sys/src/lib.rs]: why 16 here?

Comment (BusyJay) [inline grpc-sys/src/lib.rs]: See https://github.com/pingcap/grpc-rs/pull/28/files/5e2685877e0f7e37a3f810a37d27465a25c10d83#diff-da4106dc6927471f3d347e9f3d756519R67

File: src/credentials.rs [hunk 2]
```diff
@@ -0,0 +1,176 @@
...
+
+use std::ffi::CString;
+use std::ptr;
+
+use grpc_sys::{self, GrpcChannelCredentials, GrpcServerCredentials};
+use libc::c_char;
+
+pub struct ServerCredentialsBuilder {
+    root: Option<CString>,
+    cert_chains: Vec<*mut c_char>,
```
Comment (siddontang) [inline src/credentials.rs]: can we use CString here, and get the raw ptr in build directly?

So we don't need to use into_raw, from_raw.

Comment (BusyJay) [inline src/credentials.rs]: In that case a new vector has to be created and calls `into_raw`/`from_raw` when in build function. I prefer storing the pointer directly.

**Comment by disksing** [comment]
LGTM.

File: src/credentials.rs (hunk 2 above)
Comment (siddontang) [inline src/credentials.rs]: Got it

**Pull request merged**